### PR TITLE
feat(editor): Update `element-plus` to 2.4.3 (no-changelog)

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
This PR updates `element-plus` dependency to `2.4.3` so we can use the [`append-to`](2.4.3) attribute in Dialog component. This will be needed in order to implement new n8n grid layout where modals can appear in the content area or in body (compared to current setup where they are teleported to body by default).

There have been side side-effects to the update, so please take a look if all makes sense when reviewing:
1. Looks like there is a clash with `vueUse` version which made `v-on-click-outside` directive to stop working (but only in production builds). I removed it and switched to using it as a function or component.
2. New version of `element-plus` is using `aria` attributes differently so had to update design system snapshots accordingly
3. It also started using the `aria-controls` attribute which is dynamic id generated on each build and that broke some of the snapshot tests. I couldn't figure out better way so added a script that removes them from snapshots before running tests
4. Some e2e and unit tests were updated to work with new html

Please take a look if that makes sense and feel free to suggest a better approach here 🙏

[Latest e2e tests run](https://github.com/n8n-io/n8n/actions/runs/10448739633)

## Related Linear tickets, Github issues, and Community forum posts
Closes [ADO-2365](https://linear.app/n8n/issue/ADO-2365)

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/design-system/package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "@fortawesome/fontawesome-svg-core": "^1.2.36",
     "@fortawesome/free-solid-svg-icons": "^5.15.4",
     "@fortawesome/vue-fontawesome": "^3.0.3",
-    "element-plus": "^2.3.6",
+    "element-plus": "2.4.3",
```
Comment (netroy) [inline packages/design-system/package.json:47]: does it make sense to move this upgrade into a separate PR?

File: packages/design-system/src/components/N8nCheckbox/__tests__/__snapshots__/Checkbox.spec.ts.snap [hunk 2]
```diff
@@ -10,7 +10,6 @@ exports[`components > N8nCheckbox > should render with both child and label 1`]
       class="el-checkbox__input"
     >
       <input
-        aria-hidden="false"
```
Comment (netroy) [inline packages/design-system/src/components/N8nCheckbox/__tests__/__snapshots__/Checkbox.spec.ts.snap:13]: can we keep update the components to ensure that we don't lose these accessibility attributes, instead of removing them from the test snapshot 🙏🏽

**Review by cstuncsik — DISMISSED** [review]

File: packages/design-system/package.json (hunk 1 above)
Comment (MiloradFilipovic) [reply packages/design-system/package.json:47]: Not sure here, the only changes left that are unrelated to this upgrade are few lines of css in [this file](https://github.com/n8n-io/n8n/pull/10281/files#diff-f9e39ebedc3435e839605b9e7c432fc08bb554caa5dc7bcd5751c9007d762282), so I would keep it together if there are no major objections.

File: packages/design-system/src/components/N8nCheckbox/__tests__/__snapshots__/Checkbox.spec.ts.snap (hunk 2 above)
Comment (MiloradFilipovic) [reply packages/design-system/src/components/N8nCheckbox/__tests__/__snapshots__/Checkbox.spec.ts.snap:13]: I don't think this is in our control since this is the attribute added by the `element-plus` library. Looking at [the docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden), `false` for this attribute doesn't do much (will not even re-expose the element to assistive technology if any of its parents are hidden).

**Review by netroy — DISMISSED** [review]

File: cypress/e2e/2111-ado-support-pinned-data-in-expressions.cy.ts [hunk 3]
```diff
@@ -124,12 +124,12 @@ describe('ADO-2111 expressions should support pinned data', () => {
...
 		ndv.getters.inputLabel().eq(0).click();
 
-		ndv.getters.parameterInput('value').eq(1).click();
+		ndv.getters.parameterInput('value').eq(1).click({ force: true });
 		ndv.getters
 			.inlineExpressionEditorOutput()
 			.should(
 				'have.text',
-				'0,0[Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute previous nodes for preview][Execute previous nodes for preview][Execute previous nodes for preview]',
+				'[Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute node ‘PinnedSet’ for preview][Execute previous nodes for preview][Execute previous nodes for preview][undefined]',
```
Comment (netroy) [inline cypress/e2e/2111-ado-support-pinned-data-in-expressions.cy.ts:132]: this change looks unrelated

**Review by netroy — CHANGES_REQUESTED** [review]

**Review by netroy — APPROVED** [review]
Chromatic found no major visual regression between this branch and master 🎉

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.56.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.56.0)
